src/turn_integrator.py

```diff
@@ -85,8 +88,84 @@ def send_media_message(msisdn, media_type, media_id, line_name, caption=""):
             "caption": caption
         }
 
-    message_data["recipient_type"] = "individual"
-    message_data["to"] =  msisdn
+    response = send_message(message_data, line_name)
+    print(response.text)
+    return response
+
+
+### Send an interactive message with a dropdown or buttons.
+### The sections argument is a dictionary with a few attrbutes:
+### 'header_text' - text, optional
+### 'header_image' - id of a saved image, optional
+### 'body_text' - text, required
+### 'footer_text' - text, optional
+### 'buttons' - array, required array of buttons with text and callback_id, used if the interactive_type
+### is a 'button'.
+### 'list_button' - text, required text of the list button for interactive_type = 'list'
+### 'list_title' - text, required title text for the list for interactive_type = 'list'
+### 'list_items' - array, required array of items with text and callback_id for interactive_type = 'list'
+### Further details about the API call here:
+### https://whatsapp.turn.io/docs/api/messages#interactive-messages
+
+def send_interactive_message(msisdn, interactive_type, line_name, sections):
+    message_data = {
+        "to": msisdn,
+        "type": "interactive",
+        "interactive": {
+            "type": interactive_type,
+            "body": {
+                "text": sections['body_text']
+            },
+            'action': {}
+        },
+    }
+
+    if sections['header_text']:
+        message_data['interactive']['header'] = {
+            "type": "text",
+            "text": sections['header_text']
+        }
+    elif sections['header_image']:
+        message_data['interactive']['header'] = {
+            "type": "image",
+            "id": sections['header_image']
+        }
+
+    if sections['footer_text']:
+        message_data['interactive']['footer'] = {
+            "text": sections['footer_text']
+        }
+
+    if interactive_type == 'button':
+        message_data['interactive']['action']['buttons'] = []
+        for button in sections['buttons']:
+            message_data['interactive']['action']['buttons'].append(
+                {
+                    "type": "reply",
+                    "reply": {
+                        "id": button['callback_id'],
+                        "title": button['text']
+                    }
+                }
+            )
+
+    if interactive_type == 'list':
+        message_data['interactive']['action']['button'] = sections['list_button']
+        message_data['interactive']['action']['sections'] = []
+        message_data['interactive']['action']['sections'].append({})
+        message_data['interactive']['action']['sections'][0]['title'] = sections['list_title']
+        message_data['interactive']['action']['sections'][0]['rows'] = []
```

Comment:
I think probably the best way to improve this would be to use objects instead of manipulating the json directly? Using something like pydantic for objects and validation of them?